### DISCUSSION TITLE
When and how to reach out to a contributor to become a fellow maintainer?

### DISCUSSION THREAD

**Discussion by tim-schilling** [carltongibson/notes#31] [Being an open source maintainer]
I'm curious how others have handled this in their projects of varying size.

**Comment by carltongibson** [comment]
Good question. 

I've not been perfect at this over the years. At some point though I kind of realised (maybe obvious) that the upsides of inviting people outweigh the concerns. 

First the concern: you have to hold back release rights until you really know someone. You also have to make sure you can review what goes in. If that's not the case what follows doesn't apply. 

Worst case is basically, someone makes a mistake on main, and you have to revert or correct it. 

In general that doesn't actually happen. 

So on crispy-forms and channels I adopted the policy of inviting people who'd taken a regular interest.

Not all of those stayed. But some did.

**Reply by tim-schilling** [reply]
I like the idea of inviting people who take a regular interest. For those who didn't stay, do you end up revoking their permissions after some time?

**Reply by carltongibson** [reply]
@tim-schilling Yes... — like not as a matter of urgency, but some time later when I'm in there, I will just remove them as collaborators.

**Reply by tim-schilling** [reply]
That makes sense. Thanks for sharing! You hit the nail on the head originally. There's a concern about bringing people in too early.
